### ISSUE TITLE
False positive for KeyFor checker with long literal

### ISSUE THREAD

**Issue by wmdietlGC** [typetools/checker-framework#1399]
Take the following simple program and run the Nullness Checker:

```java
class Box<T> {
  static <T> Box<T> box(Class<T> type) {
    return new Box<T>();
  }
  void act(T instance) {}
}

abstract class Math {
  abstract <T> Box<T> id(Box<T> in);

  void foo(Math m) {
    m.id(Box.box(Long.class)).act(10L);
  }
}
```

The output is:

```
Math.java:12: error: [argument.type.incompatible] incompatible types in argument.
    m.id(Box.box(Long.class)).act(10L);
                                  ^
  found   : @UnknownKeyFor long
  required: ?[ extends @UnknownKeyFor Object super @KeyForBottom Void]
1 error
```

This occurs with master and an older version of #1387. Haven't tried with newest version of #1387, but wanted to report it for the test case.
